src/Symfony/Component/Messenger/MessageRecorder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Component\Cache\ResettableInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class MessageRecorder implements MessageRecorderInterface, RecordedMessageContainerInterface, ResettableInterface
+{
+    private $messages = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRecordedMessages(): array
```

Comment:
I am not fan of `getter`, it does not follow the `tell, don't ask` principle. IMOH `recordedMessages` is enough/better :)

Reply:
As far as I know, this is the "symfony way". Im not sure if I can apply "`tell, don't ask` principle". Even though we may like it more at the moment.

Reply:
ok I understand :(